### ISSUE TITLE
Migration from javax to jakarta in Glassfish. Dependency issues.

### ISSUE THREAD

**Issue by jbescos** [eclipse-ee4j/glassfish#22892]
There are some dependency issues that makes difficult the migration from javax to jakarta in a progressive way. For example, lets suppose we migrate jsonp from javax to jakarta packages and you want to include it in Glassfish:

Glassfish will throw NoDefClassFound and ClassNotFound  errors, because other libraries depends on Jsonp and they reference the classes with 'javax' instead of 'jakarta' (for example Jersey depends on Jsonp).

We don't run into this problem if we migrate all the libraries together, but this will need more time.

Do you have any recommendation about how to proceed with this migration?.

**Comment by smillidge** [comment]
I assume we will need to integrate in waves like the api has been but likely in reverse?

**Comment by m0mus** [comment]
@smillidge I don't think that it makes sense. Implementations dependency trees are much more complex. There are also a couple of cross-dependencies. It means that it will become an X step process. 
IMO, the best approach is to dedicate `master` to this work assuming that it won't be buildable for some time. Alternative is to dedicate a branch for it and merge it into master when work is done. Because the change is huge and will take some time there could be problems with rebasing. It's why I prefer the first option.
We are ready to integrate some jakartified implementations to GF and looking for guidance from you folks about how shall we do it.

**Comment by smillidge** [comment]
We moved master to 6.0.0-SNAPSHOT so feel free to make PRs against master.

**Comment by bshannon** [comment]
Lukas and I decided to try to do this on a branch.  I'm going to migrate GlassFish
to the new versions of Jakarta Activation and Jakarta Mail, then turn the branch
over to him to migrate Jakarta XML Web Services, Jakarta XML Binding, etc.
Once we got the branch working with those, we would merge it with master.
We're not very far along on this, so I have no idea how well it's going to work.
I just hate to leave master broken for how long it's going to take us to do it.
